app/modules/shelves/components/shelf_box.svelte

```diff
@@ -6,23 +6,38 @@
   import ShelfEditor from '#shelves/components/shelf_editor.svelte'
   import Modal from '#components/modal.svelte'
   import { onChange } from '#lib/svelte/svelte'
+  import { createEventDispatcher, getContext, tick } from 'svelte'
+  import { debounce } from 'underscore'
 
   export let shelf, withoutShelf
 
-  let itemsCount
+  let itemsCount, shelfBoxEl
 
-  let name, description, picture, iconData, iconLabel, isEditable
+  let name, description, picture, iconData, iconLabel, isEditable, pathname
   function refreshData () {
-    ;({ name, description, picture, iconData, iconLabel, isEditable } = serializeShelfData(shelf, withoutShelf))
+    ;({ name, description, picture, iconData, iconLabel, isEditable, pathname } = serializeShelfData(shelf, withoutShelf))
   }
   $: onChange(shelf, refreshData)
 
   let showShelfEditor = false
   const addItems = () => app.execute('add:items:to:shelf', shelf)
-  const closeShelf = () => app.vent.trigger('close:shelf')
+
+  const dispatch = createEventDispatcher()
+  const closeShelf = () => dispatch('closeShelf')
+
+  const focusStore = getContext('focus-store')
+
+  async function onFocus () {
+    if (!shelfBoxEl) await tick()
+    app.navigate(pathname, { pageSectionElement: shelfBoxEl })
+  }
+
+  const debouncedOnFocus = debounce(onFocus, 500, true)
+
+  $: if ($focusStore.type === 'shelf') debouncedOnFocus()
 </script>
 
-<div class="shelf-box">
+<div class="shelf-box" bind:this={shelfBoxEl}>
```

Comment:
somehow `shelf-box` lost its background color, making is difficult to understand its limit when filters are wrapped
![2023-04-03_12-28](https://user-images.githubusercontent.com/5363918/229484461-4a5b330f-fe51-424d-af7c-8761b8df3129.png)

Reply:
fixed in 9ed203d4f